FieldOpt/Reservoir/grid/cell.cpp

```diff
@@ -57,30 +58,76 @@ bool Cell::EnvelopsPoint(Eigen::Vector3d point) {
     for (Face face : faces_) {
         double dot_prod = (point - face.corners[0]).dot(face.normal_vector);
         if ( dot_prod < 0)
+        {
             point_inside = false;
+            break;
+        }
     }
     return point_inside;
 }
 
-void Cell::initializeFaces() {
-    int face_indices_points[6][4] = {
-        {0, 2, 1, 3},
-        {4, 5, 6, 7},
-        {0, 4, 2, 6},
-        {1, 3, 5, 7},
-        {0, 1, 4, 5},
-        {2, 6, 3, 7}
-    };
-
-
-//    int face_indices_points[6][4] = {
-//        {2, 0, 3, 1},
-//        {6, 7, 4, 5},
-//        {2, 6, 0, 4},
-//        {3, 1, 7, 5},
-//        {2, 3, 6, 7},
-//        {0, 4, 1, 5}
-//    };
+constexpr const std::array<std::array<int,4>, 6> Cell::_faces_definition_earth_pointing_z;
+constexpr const std::array<std::array<int,4>, 6> Cell::_faces_definition_ski_pointing_z;
+
+void Cell::initializeFaces()
+{
+    // The code assumes the corners of the cell are given in the following order
+    //
+    //     Above:
+    //     2---3
+    //     |   |
+    //     0---1
+    //
+    //     Below:
+    //     6---7
+    //     |   |
+    //     4---5
+    //
+    // Above and below here refer to the relative position of the two cell faces
+    // with normal vectors less than 90 degrees relative to the axis from the center
+    // of the Earth. Here "Above" refers to the face that is farther away from the
+    // Earth center, while "Below" refers to the face closer to the center of the
+    // Earth.
+    //
+    // Reservoir grids can be either left- or right-handed. If left-handed, the
+    // direction of increasing z will point down towards the center of the Earth,
+    // while if right-handed, the direction of increasing z will be away from the
+    // center of the Earth.
+
+    // Here we determine whether the cell is left- or right-handed, which will
+    // later determine whether the vertex indices sets, (0-1-2-3) and (4-5-6-7),
+    // are assigned as "Above" or "Below" cell faces.
+
+    // Make sure there is a good definition of inside and outside of the cell.
+
+    std::array<std::array<int,4>, 6> face_indices_points;
+    if (corners_[0].z() < corners_[4].z()) // z-axis grows towards Earth's center (left-handed)
+    {
+        face_indices_points = _faces_definition_earth_pointing_z;
+    }
+    else // z-axis grows towards the ski - right-handed
+    {
+        face_indices_points = _faces_definition_ski_pointing_z;
```

Comment:
ahh, you mean _faces_definition_earth_pointing_z? what is the correct convention in this case?